### PR TITLE
Feature/190 rule loop detection 2

### LINKED ISSUE #190: Rule execution loops detection
Use the Fiware-Correlator header to detect execution loops (i.e. a rule which action is an action on CB that triggers a notification that triggers the same rule).

When that happens, the rule should be disabled an alarm raised in the logs.

### PR DESCRIPTION
**Pull request by crbrox**
@fgalan 
closes #190

### CODE REVIEW

File: documentation/logs.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Alarm ID | Severity | Description | Action
 [CHECK_MASTER](#check_master)|Major|A "slave" perseo has lost visibility of master.|Check the configured SMTP Server is accessible and working properly.
 [DATABASE](#database)|Critical|There is a problem in connection to DB. It can be the perseo database or the Orion database (accessed by no-signal checker|Check configured mongoDB is up and running and is accessible from perseo. Check that databases exist in mongoDB.
 [AUTH](#auth)|Major|There is a problem in connection to Keystone. Update-actions to Orion through PEP are not working|Check HTTP connectivity to Keystone. Check provisioned user and roles/grants.
-
+[LOOP](#loop)|Major|Some rules can be provoking an infinite loop of triggered actions|Report to client/product about possible loop with the pointed rule. Check log for the correlator in teh log message.
```
Comment (fgalan) [inline documentation/logs.md]: teh -> the

File: documentation/logs.md [hunk 2]
```diff
@@ -228,5 +228,18 @@ ____
...
+
+**Severity**: Major
+
+**Detection strategy:** `lvl`:`ERROR` `msg`:`check infinite loop`
+
+**Stop condition**: `N/A`
+
+**Description**: Some rules can be provoking an infinite loop of triggered actions.
+
+**Action**: Report to client/product about possible loop with the pointed rule. Check log for the correlator in teh log message
```
Comment (crbrox) [inline documentation/logs.md]: fixed 9281078

File: documentation/logs.md (hunk 1 above)
Comment (crbrox) [inline documentation/logs.md]: fixed fd7ea51

File: documentation/logs.md [hunk 3]
```diff
@@ -74,7 +74,7 @@ Alarm ID | Severity | Description | Action
 [CHECK_MASTER](#check_master)|Major|A "slave" perseo has lost visibility of master.|Check the configured SMTP Server is accessible and working properly.
```
Comment (fgalan) [inline documentation/logs.md]: This loop-protection functionality is nice at I think it should be described in the "main" documentation, not only in the piece about logs and alarms.

Maybe as an aditional short section after this one https://github.com/telefonicaid/perseo-fe/blob/master/documentation/architecture.md#action, if you don't think it is a better place.

File: lib/routes/actionsController.js [hunk 4]
```diff
@@ -23,21 +23,61 @@
...
+        // try detect loop
+        corrAge[correlator] = now;
+        rulesWithCorr = recentAxn[correlator];
+        if (!rulesWithCorr) { //There is no array yet for this correlator
+            rulesWithCorr = recentAxn[correlator] = [];
+        }
+        if (util.isArray(rulesWithCorr)) {
+            for (var i = 0; i < rulesWithCorr.length; i++) {
+                if (rulesWithCorr[i] === firedEvent.ruleName) {
+                    errMsg = util.format('check infinite loop for %s (%s)', firedEvent.ruleName, correlator);
```
Comment (fgalan) [inline lib/routes/actionsController.js]: What about?

```
'check infinite loop for %s (%s) - the rule has not ben triggered'
```

(replace "-" by the char you typically use to separate sensences within the same log `msg`)

Comment (crbrox) [inline lib/routes/actionsController.js]: fixed 621e264

File: documentation/logs.md (hunk 3 above)
Comment (crbrox) [inline documentation/logs.md]: check cc6d3f8

**Comment by fgalan** [comment]
LGTM

**Pull request merged**